packages/dds/merge-tree/src/collections/list.ts

```diff
@@ -3,142 +3,127 @@
  * Licensed under the MIT License.
  */
 
```

Comment:
this file is best view in split most, or view the file directly, as it is a complete re-write